ci/checks/copyright.py

```diff
@@ -13,45 +13,43 @@
 # limitations under the License.
 #
 
-import datetime
-import re
 import argparse
-import io
+import datetime
 import os
+import re
 import sys
 
-SCRIPT_DIR = os.path.dirname(os.path.realpath(os.path.expanduser(__file__)))
-
-# Add the scripts dir for gitutils
-sys.path.append(os.path.normpath(SCRIPT_DIR))
-
-# Now import gitutils. Ignore flake8 error here since there is no other way to
-# set up imports
-import gitutils  # noqa: E402
+import git
 
 FilesToCheck = [
     re.compile(r"[.](cmake|cpp|cu|cuh|h|hpp|sh|pxd|py|pyx)$"),
     re.compile(r"CMakeLists[.]txt$"),
     re.compile(r"CMakeLists_standalone[.]txt$"),
     re.compile(r"setup[.]cfg$"),
     re.compile(r"[.]flake8[.]cython$"),
-    re.compile(r"meta[.]yaml$")
+    re.compile(r"meta[.]yaml$"),
+]
+ExemptFiles = [
+    re.compile(r"cpp/include/cudf_test/cxxopts.hpp"),
+    re.compile(r"versioneer[.]py"),
 ]
-ExemptFiles = ["cpp/include/cudf_test/cxxopts.hpp"]
 
 # this will break starting at year 10000, which is probably OK :)
 CheckSimple = re.compile(
-    r"Copyright *(?:\(c\))? *(\d{4}),? *NVIDIA C(?:ORPORATION|orporation)")
+    r"Copyright *(?:\(c\))? *(\d{4}),? *NVIDIA C(?:ORPORATION|orporation)"
+)
 CheckDouble = re.compile(
     r"Copyright *(?:\(c\))? *(\d{4})-(\d{4}),? *NVIDIA C(?:ORPORATION|orporation)"  # noqa: E501
 )
 
 
 def checkThisFile(f):
-    # This check covers things like symlinks which point to files that DNE
-    if not (os.path.exists(f)):
-        return False
-    if gitutils and gitutils.isFileEmpty(f):
+    if isinstance(f, git.Diff):
```

Comment:
This is a little awkward -- this now allows passing file paths OR `git.Diff` objects which allow us to check against the git stage rather than the contents on disk. Checking against the git stage is necessary for correctly checking against what will be committed. However, I didn't want to remove the previous functionality (file paths) which is used when `--git-modified-only` is not passed.

Reply:
I'll try to review this in the next day or two and see what I come up with.